testsuite/model/src/test/java/org/keycloak/testsuite/model/session/OfflineSessionPersistenceTest.java

```diff
@@ -160,6 +163,32 @@ public void testPersistenceMultipleNodesClientSessionAtSameNode() throws Interru
                 }).forEach(userSessionModel -> clientSessionIds.add(userSessionModel.getId())));
                 return null;
             });
+
+            // ensure that all session have been created on all nodes
+            intermediate.countDown();
+            try {
+                intermediate.await();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException(e);
+            }
+
+            // defer the shutdown and check if all sessions exist to ensure that they replicate across the different nodes
+            // this should avoid an "org.infinispan.remoting.transport.jgroups.SuspectException: ISPN000400: Node node-XX was suspected"
+            while (true) {
+                try {
+                    assertOfflineSessionsExist(realmId, clientSessionIds);
+                    break;
+                } catch (AssertionError e) {
+                    log.warn("assertion failed, retrying to see if all sessions exist.");
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException ex) {
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException(e);
+                    }
```

Comment:
This occurs in several places. Could this be extracted into a method, e.g. `interruptibleSleep(int timeMs)`

Reply:
Done. I named it "sleep" as there is no other sleep around to make it more specific. Please re-review.